docs/source/Getting-Started/Tutorials/Build-Enclave-App-Hello-World.rst

```diff
@@ -0,0 +1,257 @@
+
+The script will compile the eapp, the runtime, and the host.
+Take a look at the first few lines of the script.
+In most cases, changing the variables in the script will be sufficient to build any custom enclaves.
+
+.. code-block:: bash
+
+	################################################################
+	#                   Replace the variables                      #
+	################################################################
+	NAME=hello
+	VAULT_DIR=`dirname $0`
+	BUILD_COMMAND="make -C eapp && make -C host"
+	OUTPUT_DIR=$KEYSTONE_SDK_DIR/../buildroot_overlay/root/$NAME
+	EYRIE_DIR=$KEYSTONE_SDK_DIR/rts/eyrie
+	EYRIE_PLUGINS="freemem untrusted_io_syscall linux_syscall env_setup"
+	PACKAGE_FILES="eapp/hello \
+	               host/runner \
+	               $EYRIE_DIR/eyrie-rt"
+	PACKAGE_SCRIPT="./runner hello eyrie-rt"
+
+We will go through what each variable is for, and how the script builds the enclave.
+
+``NAME`` defines the name of the enclave. It will be used to generate the final enclave package.
+
+``VAULT_DIR`` is the path to the script. The script will change the directory to ``$VAULT_DIR``
+before running the build commands.
+
+``BUILD_COMMAND`` defines the build commands. This example has ``Makefile`` in each of the
+directory, so the script just needs to execute ``make`` at each directory to build both the eapp and
+the host.
+
+``OUTPUT_DIR`` is the path to the output files. Since we started from :doc:`Quick Start
+<../Running-Keystone-with-QEMU>`, we put the build outputs to the buildroot overlay directory so
+that we can see them in the QEMU disk image.
+
+``EYRIE_DIR`` is the path to the Eyrie runtime source code.
+
+``EYRIE_PLUGINS`` defines what plugins you want to include in the runtime.
+
+``PACKAGE_FILES`` defines which files you want to include in the final enclave package. 
+``vault.sh`` uses `Makeself <https://makeself.io/>`_ to generate a self-extracting archive for the
+enclave.
+The package usually includes the host binary (i.e., ``host/runner``), the eapp binary (i.e.,
+``eapp/hello``), and the runtime binary (i.e., ``$EYRIE_DIR/eyrie-rt``)
+
+``PACKAGE_SCRIPT`` defines the input command to ``makeself``. The self-extracting archive will
+execute this command after the decompression.
+
+Eyrie Runtime
+------------------------------
+
+If you have completed :doc:`Quick Start <../Running-Keystone-with-QEMU>`, the runtime source code
+would have been already located at ``$EYRIE_DIR``.
+
+You can find the following command in ``vault.sh``, which builds the Eyrie runtime.
+
+::
+
+	$EYRIE_DIR/build.sh $EYRIE_PLUGINS
+
+
+Enclave Application: hello.c
+------------------------------
+ 
+Open ``hello.c`` file in ``sdk/exmamples/hello/eapp/``. This is the source code of the enclave
+application.
+
+.. code-block:: c
+
+	#include <stdio.h>
+
+	int main()
+	{
+	  printf("hello, world!\n");
+	  return 0;
+	}
+
+This is the standard C program that we will run isolated in an enclave.
+
+Host Application: host.cpp
```

Comment:
+ It may be better to _not_ include the code of the sample apps in
   the doc.
   - Any edits don't need to be copied, less confusion.
   - Makes them open the files and read em.
   - Rather than point them to the definition path for params, point
     to a doc page that has technical documentation for params.
   - In tutorial code I'd avoid putting multiple function calls on one
     line. (shared buffer)